bofire/strategies/predictives/botorch.py

```diff
@@ -114,6 +102,21 @@ def _features2names(self) -> Dict[str, Tuple[str]]:
         )
         return features2names
 
+    def _get_optimizer_options(self) -> Dict[str, int]:
+        """Returns a dictionary of settings passed to `optimize_acqf` controlling
+        the behavior of the optimizer.
+
+        Returns:
+            Dict[str, int]: The dictionary with the settings.
+        """
+        return {
+            "batch_limit": self.batch_limit
+            if len(self.domain.constraints.get([NChooseKConstraint, ProductConstraint]))
+            == 0
+            else 1,  # type: ignore
```

Comment:
Why `type: ignore`?

Reply:
Due to the type hints in the data model for `batch_limit` needed.